client/signup/config/steps-pure.js

```diff
@@ -401,6 +401,11 @@ export function generateSteps( {
 			stepName: 'site-information',
 			providesDependencies: [ 'siteTitle', 'address', 'email', 'phone' ],
 		},
+
+		'site-style': {
+			stepName: 'site-style',
+			providesDependencies: [ 'themeSlugWithRepo' ],
```

Comment:
Does this step need an action?

Reply:
That's not what I mean :)

I'm thinking about how this step fits into the framework as a whole. What would happen if I wanted to use this step in a different flow outside of onboarding? We should be trying to build steps the can be easily added/removed from different flows.

Reply:
> I'm thinking about how this step fits into the framework as a whole ...  We should be trying to build steps the can be easily added/removed from different flows.

I agree with the sentiment. Is there something particular you had in mind? Apologies if you have to cover old ground, but it'd help to have a more specific explanation in order that we may improve this component. 

This PR creates the component only; an integration task will follow. There's a strict dependency on `siteType`, which I see as a barrier to flow-independence. Perhaps if we added a fall back set of [site style options](https://github.com/Automattic/wp-calypso/pull/29019/files#diff-740be341c75a521ad5583216809076faR8) with themes that did not rely on `siteType` being present? Would that address this concern?

cc @michaeldcain 

> Does this step need an action?

Could explain what you mean by "action"? 

Thanks again!

Reply:
> There's a strict dependency on siteType

This is true, albeit temporary, as we eventually scale the preview to work with any segment. To get the step in place, I'm fine with the `siteType` constraint.

`siteTopic` however is not a requirement, as we have a "master" business template that we can fall back to. In the design and early iterations, we showed this initial site as soon as a user picked the "business" segment. @shaunandrews should we return to showing it earlier/when a vertical without a parent is chosen?

Regardless of what's shown on the `business-site-topic` step, I think we can use the master business template as a fallback to remove that dependency from the site-style step.

Reply:
> Could explain what you mean by "action"?

I don't want to speak for @scruffian, but I think he means a step action to help modularize the step (i.e. an endpoint call to set the theme on the site in this case). This is harder to do with our single `/sites/new` call approach that Signup currently takes, but there is a method to defer the action until the end of Signup. That being said, I worry that it would cause an issue with how Headstart is run on the new site endpoint (and requires a theme to properly set up).

I'd say that we attempt to make the steps as independent as possible in the current project scope, but that we avoid refactoring the way Signup currently creates and sets up a new site until our maintenance phase after Phase 1 launches.

Reply:
> step action to help modularize the step (i.e. an endpoint call to set the theme on the site in this case)

That makes sense. 👍 

Thanks @michaeldcain and @scruffian,  I'll make another pass on the component with this discussion in mind

Reply:
> In the design and early iterations, we showed this initial site as soon as a user picked the "business" segment.

It's the `is-empty` [class assignment](https://github.com/Automattic/wp-calypso/blob/1b5d4e0/client/signup/site-mockup/index.jsx#L65)that needs to be either removed or modified:

``
		const siteMockupClasses = classNames( {
			'site-mockup__wrap': true,
			'is-empty': isEmpty( this.props.verticalData ),
		} );
```

Reply:
> I assume the "master", default, business theme (i.e., the one we fall back to when there is no vertical) will also have theme style variations?

@ramonjd that's my fault for the overlapping taxonomy here. The master business template is just a set of blocks inside the Radcliffe theme, _not_ a different theme. Thus, the styles are the same as the ones included in Radcliffe. I was just saying that there is fallback content to be displayed if a site topic hasn't been set.

Reply:
> I don't want to speak for @scruffian,

You said what I was thinking. :)